src/api/ucc_status.h

```diff
@@ -0,0 +1,35 @@
+/**
+ * @file ucc_status.h
+ * @date 2020
+ * @copyright Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_STATUS_H_
+#define UCC_STATUS_H_
+
+
+/**
+ * @brief Status codes for the UCC operations
+ */
+
+typedef enum {
+    /* Operation completed successfully */
+    UCC_OK                              =   0,
+
+    /* Operation is posted and is in progress */
+    UCC_INPROGRESS                      =   1,
```

Comment:
Do we intend to use this? according to UCX semantics, collective_post() returns UCC_OK if the operation is completed (no need to call test()) and UCC_INPROGRESS if it started successfully...

Reply:
Given that collectives are relatively high-latency operations, we should expect to use it.

Reply:
This opens up several questions about completion. UCX has send_nb(), send_nbr() and send_nbx(). The differences are related to the way the request is allocated and filled as successful (there's some kind of race condition between creation and completion, but I don't have all the details). We should learn from their experience...

Reply:
I misread your question. I thought you are asking just about UCC_INPROGRESS as it was pointing to line 22. 

The semantics that you describe for UCC_OK is not supported (or doesn't have parallel) in UCC .i.e., there is no implicit creation and destruction of the request object. The request object is always created. Remember, in UCC, the collective operation is initialized with the ucc_collective_init interface. So before calling ucc_collective_post the request object is already created. The post is really just invoking the operation. So, post is returning UCC_OK or UCC_INPROGRESS based on the operation’s status. The request is finalized only after calling ucc_collective_finalize interface. 

I will put this on tomorrow's agenda. So, if it is not clear, we can discuss it.

Reply:
Aug 19th WG Discussion: 
* Will not support implicit creation and destruction of the request object. 
* Discuss with MPICH team, if they need request object to be created by the MPICH rather than UCC

Reply:
Sep 9th WG discussion:

UCC currently requires the programming models to create and destroy the request. There is no implicit request object creation and destroying. During the WG discussion, Ken from the MPICH team was convinced that having explicit request API is helpful and enough for their purposes .i.e., they will hold the pointer of the request object in their (MPICH) request object. So, no changes are required here.